dev-lang/zig/zig-0.4.0.ebuild

```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit cmake-utils
+
+DESCRIPTION="A robust, optimal, and maintainable programming language."
+HOMEPAGE="https://ziglang.org/"
+LICENSE="MIT"
+SLOT="0"
+IUSE=""
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://github.com/ziglang/zig.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/ziglang/zig/archive/${PV}.tar.gz -> zig-${PV}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+ALL_LLVM_TARGETS=( AArch64 AMDGPU ARM BPF Hexagon Lanai Mips MSP430 NVPTX
+	PowerPC Sparc SystemZ WebAssembly X86 XCore )
+ALL_LLVM_TARGETS=( "${ALL_LLVM_TARGETS[@]/#/llvm_targets_}" )
+# According to zig's author, zig builds that do not support all targets are not
+# supported by the upstream project.
+LLVM_TARGET_USEDEPS=${ALL_LLVM_TARGETS[@]}
```

Comment:
Have you tried and confirmed this? This is really scary and to me doesnt make any sense enabling all targets. I'd personally add a some sort of 'experimental' USE flag where, if it works, you could run this without everything enabled. Then just print some info message that upstream support, and maybe support altogether, isnt offered due to experimental flag. 

I dont have answers to this, just looks a bit.. suspicious and scary :)

Reply:
If zig is used to compile for an architecture that the installed llvm does not support, it gives a misleading error message:
```
$ zig build-exe -target powerpc-linux-gnu hello.zig
Zig is expecting LLVM to understand this target: 'powerpc-unknown-linux-gnu'
However LLVM responded with: "No available targets are compatible with triple "powerpc-unknown-linux-gnu""
Zig is unable to continue. This is a bug in Zig:
https://github.com/ziglang/zig/issues/438
```
I've added an experimental use flag as you suggested.